src/mlpack/methods/ann/layer/isrlu.hpp

```diff
@@ -0,0 +1,169 @@
+ *
+ * @tparam InputDataType Type of the input data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ * @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ */
+template <
+    typename InputDataType = arma::mat,
+    typename OutputDataType = arma::mat
+>
+class ISRLU
+{
+ public:
+  /**
+   * Create the ISRLU object using the specified parameters.
+   * The non zero gradient can be adjusted by specifying the parameter
+   * alpha in the range 0 to 1. Default (alpha = 0.03)
+   *
+   * @param alpha Non zero gradient
+   */
+  ISRLU(const double alpha = 0.03);
+
+  /**
+   * Ordinary feed forward pass of a neural network, evaluating the function
+   * f(x) by propagating the activity forward through f.
+   *
+   * @param input Input data used for evaluating the specified function.
+   * @param output Resulting output activation.
+   */
+  template<typename InputType, typename OutputType>
+  void Forward(const InputType&& input, OutputType&& output);
+
+  /**
+   * Ordinary feed backward pass of a neural network, calculating the function
+   * f(x) by propagating x backwards through f. Using the results from the feed
+   * forward pass.
+   *
+   * @param input The propagated input activation.
+   * @param gy The backpropagated error.
+   * @param g The calculated gradient.
+   */
+  template<typename DataType>
+  void Backward(const DataType&& input, DataType&& gy, DataType&& g);
+
+  //! Get the output parameter.
+  OutputDataType const& OutputParameter() const { return outputParameter; }
+  //! Modify the output parameter.
+  OutputDataType& OutputParameter() { return outputParameter; }
+
+  //! Get the delta.
+  OutputDataType const& Delta() const { return delta; }
+  //! Modify the delta.
+  OutputDataType& Delta() { return delta; }
+
+  //! Get the non zero gradient.
+  double const& Alpha() const { return alpha; }
+  //! Modify the non zero gradient.
+  double& Alpha() { return alpha; }
+
+  /**
+   * Serialize the layer.
+   */
+  template<typename Archive>
+  void serialize(Archive& ar, const unsigned int /* version */);
+
+ private:
+  /**
+   * Computes the ISRLU function
+   *
+   * @param x Input data.
+   * @return f(x).
+   */
+  double Fn(const double x)
+  {
+    if (x < 0)
+      return x / (std::sqrt(1 + alpha * x * x));
+    else
+      return x;
+  }
+
+  /**
+   * Computes the ISRLU function using a dense matrix as input.
+   *
+   * @param x Input data.
+   * @param y The resulting output activation.
+   */
+  template<typename eT>
+  void Fn(const arma::Mat<eT>& x, arma::Mat<eT>& y)
+  {
+    if (x < 0)
+      y = x / (arma::sqrt(1 + alpha * arma::pow(x, 2)));
+    else
+      y = x;
+  }
+
+  /**
+   * Computes the first derivative of the ISRLU function.
+   *
+   * @param x Input data.
+   * @return f'(x)
+   */
+  double Deriv(const double x)
+  {
+    return (x >= 0) ? 1 : std::pow
+    (1 / std::sqrt(1 + alpha * std::pow(x, 2)), 3);
```

Comment:
I would prefer it if you used same syntax everywhere either `x * x or pow(x, 2)` whichever you think is better. Also you need to add the edge case here also.